src/wallet/walletdb.cpp

```diff
@@ -790,17 +793,19 @@ DBErrors CWalletDB::LoadWallet(CWallet* pwallet)
             string strType, strErr;
             if (!ReadKeyValue(pwallet, ssKey, ssValue, wss, strType, strErr))
             {
-                // losing keys is considered a catastrophic error, anything else
-                // we assume the user can live with:
-                if (IsKeyType(strType))
+                if (strType == "networkinfo") {
+                    // example: running mainnet, but this wallet.dat is from testnet
+                    result = DB_WRONG_NETWORK;
+                } else if (result != DB_WRONG_NETWORK && IsKeyType(strType)) {
```

Comment:
I was initially concerned about this (because it leads to potentially ignoring corrupt data), but thinking through the logic, it's entirely possible that a wallet from a different network could include changes that are incompatible with our parsers. So as long as the caller treats `DB_WRONG_NETWORK` as being at least as catastrophic as `DB_CORRUPT` then I think we're fine.